test/functional/api/vim_spec.lua

```diff
@@ -81,6 +81,26 @@ describe('api', function()
     end)
   end)
 
+  describe('nvim_execute_lua', function()
```

Comment:
> If lua is used for batching and coordination, then how is call_atomic needed?

I think that's a good observation. It's usually good to have a single (powerful) way to do things. If one has both, users/script writers will often think about which is best/most efficient/... for their situation, perhaps spending needless time on this.